src/Datadog.Trace.ClrProfiler.Managed/IntegrationVersionRange.cs

```diff
@@ -0,0 +1,92 @@
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// Specifies a safe version range for an integration.
+    /// </summary>
+    public class IntegrationVersionRange
+    {
+        /// <summary>
+        /// Gets or sets the minimum major version.
+        /// </summary>
+        public ushort MinimumMajor { get; set; } = ushort.MinValue;
+
+        /// <summary>
+        /// Gets or sets the minimum minor version.
+        /// </summary>
+        public ushort MinimumMinor { get; set; } = ushort.MinValue;
+
+        /// <summary>
+        /// Gets or sets the minimum patch version.
+        /// </summary>
+        public ushort MinimumPatch { get; set; } = ushort.MinValue;
+
+        /// <summary>
+        /// Gets or sets the maximum major version.
+        /// </summary>
+        public ushort MaximumMajor { get; set; } = ushort.MaxValue;
+
+        /// <summary>
+        /// Gets or sets the maximum minor version.
+        /// </summary>
+        public ushort MaximumMinor { get; set; } = ushort.MaxValue;
+
+        /// <summary>
+        /// Gets or sets the maximum patch version.
+        /// </summary>
+        public ushort MaximumPatch { get; set; } = ushort.MaxValue;
+
+        /// <summary>
+        /// Gets or sets the MinimumMajor, MinimumMinor, and MinimumPatch properties.
+        /// Convenience property for setting target minimum version.
+        /// </summary>
+        public string MinimumVersion
+        {
+            get => $"{MinimumMajor}.{MinimumMinor}.{MinimumPatch}";
+            set
+            {
+                var parts = value.Split('.');
+                if (parts.Length > 0)
+                {
+                    MinimumMajor = ushort.Parse(parts[0]);
+                }
+
+                if (parts.Length > 1)
+                {
+                    MinimumMinor = ushort.Parse(parts[1]);
+                }
+
+                if (parts.Length > 2)
+                {
+                    MinimumPatch = ushort.Parse(parts[2]);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the MaximumMajor, MaximumMinor, and MaximumPatch properties.
+        /// Convenience property for setting target minimum version.
+        /// </summary>
+        public string MaximumVersion
```

Comment:
Corner case: I don't expect the user to use this setter twice, but what do you think about setting default values for the remaining versions? Say, if the user specifies a Major.Minor, we make sure to set the Patch to a known default value instead of leaving it as the previous value?

Reply:
Good point. I'm actually not a big fan of using `ushort.MinValue` and `ushort.MaxValue` (even though I think I suggested that originally). I think I prefer using `null` to mean "this value was not set, so there's no restriction."

Reply:
They'd have to call it twice within the attribute which I don't think is possible?
I'm happy to add a default in this setter though.

Reply:
I think the reason we decided on those values instead of null originally was to simplify all of the consuming code, which I still like.

Reply:
That said, I'm totally fine with using null and defaulting to `USHRT_MAX` in the nlohmann section,